rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -176,21 +176,21 @@ def _to_sparse_sentence_features(
             for feature in sparse_sequence_features
         ]
 
-    def _get_features_from_parsed_message(
-        self, parsed_message: Optional[Message], attributes: Set[Text]
+    def _get_features_from_lookup_table(
+        self, sub_state: SubState, attributes: Set[Text], e2e_features: E2ELookupTable,
     ) -> Dict[Text, List["Features"]]:
-        if parsed_message is None:
+        if not sub_state:
             return {}
-
-        output = defaultdict(list)
-        for attribute in attributes:
-            all_features = parsed_message.get_sparse_features(
-                attribute
-            ) + parsed_message.get_dense_features(attribute)
-
-            for features in all_features:
-                if features is not None:
-                    output[attribute].append(features)
+        output = e2e_features.lookup_features(sub_state, attributes=attributes)
+
+        # FIXME: the followign should not be necessary... but looks more closely like
+        # what was here before
```

Comment:
I hope that those all use the `get_sparse_features` methods now which filter the features list (and also concatenate the different features - which depends on the order in which we apply featurizers 😅 ) ... there should be no need to rely on the order to identify sparse/dense because of the `is_sparse` attribute.
In the classifiers, there was a point where we only grab the first feature in the list - but I think that was only for the default_label data and there should only be one feature in that list then. 
... Is there any other tf code later that assumes this order?

Reply:
taken care of by new combine functionality - haven't checked whether downstream really relies on it yet